File: go/108.go
/**
题目概述:

给你一个整数数组 nums ，其中元素已经按 升序 排列，请你将其转换为一棵 高度平衡 二叉搜索树。

高度平衡 二叉树是一棵满足「每个节点的左右两个子树的高度差的绝对值不超过 1 」的二叉树。



*/
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
// 中序遍历 左->根->右
func sortedArrayToBST(nums []int) *TreeNode {
	if len(nums)==0{
		return nil
	}
	mid:=len(nums)/2
	left:=nums[:mid]
	right:=nums[mid+1:]
	return &TreeNode{nums[mid], sortedArrayToBST(left),sortedArrayToBST(right)}
}

File: php/108.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param Integer[] $nums
     * @return TreeNode
     */
    function sortedArrayToBST($nums) {
       if(empty($nums)){
          return null;
       }
       $mid=floor(count($nums)/2);
       $left = array_slice($nums,0,$mid);
       $right = array_slice($nums,$mid+1,$mid);
       return new TreeNode($nums[$mid],$this->sortedArrayToBST($left),$this->sortedArrayToBST($right));
    }
}